### PR TITLE
Build a REST-API to add items in cart

### PR DESCRIPTION
**Pull request by chrill-christella**
### what does this PR do?

add items to buyers' cart

### Description of the task to be completed

. Add item to the buyer's cart  
.The product details (name, price, image) should be retrieved from the database  
.The updated cart should be stored in the buyer's account  
.A confirmation message should be sent to the frontend to be displayed to the buyer

### How should this be manually tested

. clone this repository  
. open it in your favorite editor  
.cd into it  
. checkout to ft-add-to-cart  
.run migrations and seeds by npm run migrate:up and npm run db:seeds respectively in your terminal  
. start the server by running npm run dev  
. make a post request in postman [http://localhost:${port}/api/cart/add/{productId}](http://localhost:${port}/api/cart/add/%7BproductId%7D) to add product into cart  
.You may also use the documented api by running your server and then in your browser http://localhost:${port}/docs
Relevant pivotal tracker user story
[#36]

### CODE REVIEW

File: src/controllers/cartController.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import {Product, cart} from "../database/models";
+
+class CartController {
+  static async addItem(req, res) {
+    try {
+      const {productId} = req.params;
+      const product = await Product.findOne({ where: { productId, available: true } });
+      if (!product) {
+        return res.status(404).json({ status: 404, message: "Product not found" });
```
Comment (oliviertech212) [inline src/controllers/cartController.js:9]: you are duplicating statuscode
try
res.status(404).json({ status: "fail", message: "Product not found" });

Comment (oliviertech212) [inline src/controllers/cartController.js]: you are duplicating status code.     check all
try
res.status(404).json({ status: "fail", message: "Product not found" });

**Review by oliviertech212 — CHANGES_REQUESTED** [review]

File: src/controllers/cartController.js
Comment (sostene12) [inline src/controllers/cartController.js:1]: - you can change that status within json

File: src/controllers/cartController.js (hunk 1 above)
Comment (chrill-christella) [reply src/controllers/cartController.js]: yeah sure thank you!

File: test/buyer.test.js
Comment (sostene12) [inline test/buyer.test.js:1]: - also the tests are failing

**Review by sostene12 — CHANGES_REQUESTED** [review]

**Comment by chrill-christella** [comment]
[Guest/Buyer should be able to add items to their cart](https://trello.com/c/LkfCZhS2/24-guest-buyer-should-be-able-to-add-items-to-their-cart)

**Review by Shumba250 — APPROVED** [review]

**Review by oliviertech212 — APPROVED** [review]

**Review by wakaflorien — COMMENTED** [review]
- make sure you resolve conflicts

**Pull request closed without merging**